### ISSUE TITLE
Cannot do multiple mutations with apollo-server

### ISSUE THREAD

**Issue by mikew** [Urigo/graphql-modules#1732]
**Describe the bug**

When running multiple mutations, graphql-modules breaks with:

```
TypeError: Cannot read property 'count' of undefined
    at getSession (:4000/Users/mikewyatt/Work/promoboxx/graphql-module-woes/dist/graphql-modules/src/application/apollo.js:40)
    at executor (:4000/Users/mikewyatt/Work/promoboxx/graphql-module-woes/dist/graphql-modules/src/application/apollo.js:47)
    at :4000/Users/mikewyatt/Work/promoboxx/graphql-module-woes/dist/delegate/src/delegateToSchema.js:86
    at new ValueOrPromise (:4000/Users/mikewyatt/Work/promoboxx/graphql-module-woes/node_modules/value-or-promise/src/ValueOrPromise.ts:33)
    at delegateRequest (:4000/Users/mikewyatt/Work/promoboxx/graphql-module-woes/dist/delegate/src/delegateToSchema.js:86)
    at Object.delegateToSchema (:4000/Users/mikewyatt/Work/promoboxx/graphql-module-woes/dist/delegate/src/delegateToSchema.js:20)
    at :4000/Users/mikewyatt/Work/promoboxx/graphql-module-woes/dist/wrap/src/generateProxyingResolvers.js:64
    at :4000/Users/mikewyatt/Work/promoboxx/graphql-module-woes/dist/wrap/src/generateProxyingResolvers.js:60
    at field.resolve (:4000/Users/mikewyatt/Work/promoboxx/graphql-module-woes/node_modules/apollo-server-core/src/utils/schemaInstrumentation.ts:103)
    at resolveField (:4000/Users/mikewyatt/Work/promoboxx/graphql-module-woes/node_modules/graphql/execution/execute.js:464)
```

**To Reproduce**

```graphql
mutation {
  exampleMutation
  anotherMutation
}
```

https://github.com/mikew/grapql-module-woes

**Expected behavior**

Both mutations should run [as per the spec](http://spec.graphql.org/June2018/#sec-Mutation)

**Environment:**

- OS: alpine, macos
- `@graphql-modules/...`: this scope is not used anymore
- NodeJS: 12, 15

**Additional context**

**Comment by unzico** [comment]
We are having the exact same issue. Did you find any solution for this? This is effectively blocking us from upgrading to v1 (v1.4.3) of graphql-modules (we're still using v0 in production).

**Environment:**
OS: Windows 10
`graphql-modules`: v1.4.3
Node.js: v12

**Comment by mikew** [comment]
We've worked around it by making new mutations on the server and calling them from the client. It's a bummer, because it removes a huge flexibility of graphql. We also have a PR internally that strips out graphql-modules. Since we're already using apollo, it's `dataSources` concept suffices.

**Comment by laurisvan** [comment]
I am facing the same issue, too, and unfortunately we just rolled out the version migration to graphql-modules 1.0 into production without noticing this issue first. :(

In my quick research the problem appears on every multi-mutation operation, and it looks as if the session tied to the first mutation is destroyed before the next mutation is handled, and that they share the same session.

**Comment by laurisvan** [comment]
Could this problem be as simple as not deleting the context id handle when deleting the related session.

At least I was able to run the multiple mutations when adding one line as follows

```
    function getSession(ctx: any) {
      if (!ctx[CONTEXT_ID]) {
        ctx[CONTEXT_ID] = uniqueId((id) => !sessions[id]);
        const { context, ɵdestroy: destroy } = contextBuilder(ctx);

        sessions[ctx[CONTEXT_ID]] = {
          count: 0,
          session: {
            context,
            destroy() {
              if (--sessions[ctx[CONTEXT_ID]].count === 0) {
                destroy();
                delete sessions[ctx[CONTEXT_ID]];
                // This is the patched line
                delete ctx[CONTEXT_ID];
              }
            },
          },
        };
      }
```

**Comment by dotansimha** [comment]
The fix is available on `graphql-modules@1.4.4`
